webapi/src/test/scala/org/knora/webapi/responders/admin/ListsResponderADMSpec.scala

```diff
@@ -1021,5 +1021,46 @@ class ListsResponderADMSpec extends CoreSpec(ListsResponderADMSpec.config) with
         response.canDeleteList should be(true)
       }
     }
+
+    "used to delete list node comments" should {
+      "delete all comments of child node that contains just one comment" in {
+        val nodeIri = "http://rdfh.ch/lists/0001/testList01"
+        responderManager ! ListChildNodeCommentsDeleteRequestADM(
+          iri = nodeIri,
+          featureFactoryConfig = defaultFeatureFactoryConfig,
+          requestingUser = SharedTestDataADM.anythingAdminUser
+        )
+        val response: ListChildNodeCommentsDeleteResponseADM =
+          expectMsgType[ListChildNodeCommentsDeleteResponseADM](timeout)
+        response.nodeIri should be(nodeIri)
+        response.commentsDeleted should be(true)
+      }
+
+      "delete all comments of child node that contains more than one comment" in {
+        val nodeIri = "http://rdfh.ch/lists/0001/testList02"
+        responderManager ! ListChildNodeCommentsDeleteRequestADM(
+          iri = nodeIri,
+          featureFactoryConfig = defaultFeatureFactoryConfig,
+          requestingUser = SharedTestDataADM.anythingAdminUser
+        )
+        val response: ListChildNodeCommentsDeleteResponseADM =
+          expectMsgType[ListChildNodeCommentsDeleteResponseADM](timeout)
+        response.nodeIri should be(nodeIri)
+        response.commentsDeleted should be(true)
+      }
+
+      "not delete a comment of root list node" in {
+        val nodeIri = "http://rdfh.ch/lists/0001/testList"
+        responderManager ! ListChildNodeCommentsDeleteRequestADM(
+          iri = nodeIri,
+          featureFactoryConfig = defaultFeatureFactoryConfig,
+          requestingUser = SharedTestDataADM.anythingAdminUser
+        )
+        val response: ListChildNodeCommentsDeleteResponseADM =
+          expectMsgType[ListChildNodeCommentsDeleteResponseADM](timeout)
+        response.nodeIri should be(nodeIri)
+        response.commentsDeleted should be(false)
+      }
+    }
```

Comment:
In the test data you added a node without comments. Maybe you wanted to check that case as well (which I think is important) and forgot? I would also test the case with the node being used in data. So we can ensure that nobody can delete a node that is referenced anywhere.

Reply:
Correct. Before wasn't able to use it, but now after refactor added also that test case.